amethyst_rendy/src/formats/texture.rs

```diff
@@ -0,0 +1,181 @@
+    },
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(transparent)]
+pub struct ImageFormat(pub ImageTextureConfig);
+
+impl Default for ImageFormat {
+    fn default() -> Self {
+        use rendy::{
+            hal::image::{Anisotropic, PackedColor, SamplerInfo, WrapMode},
+            texture::image::{Repr, TextureKind},
+        };
+
+        ImageFormat(ImageTextureConfig {
+            format: None,
+            repr: Repr::Srgb,
+            kind: TextureKind::D2,
+            sampler_info: SamplerInfo {
+                min_filter: Filter::Nearest,
+                mag_filter: Filter::Nearest,
+                mip_filter: Filter::Nearest,
+                wrap_mode: (WrapMode::Tile, WrapMode::Tile, WrapMode::Tile),
+                lod_bias: 0.0.into(),
+                lod_range: std::ops::Range {
+                    start: 0.0.into(),
+                    end: 8000.0.into(),
+                },
+                comparison: None,
+                border: PackedColor(0),
+                anisotropic: Anisotropic::Off,
+            },
+            generate_mips: false,
+        })
+    }
+}
+
+amethyst_assets::register_format_type!(TextureData);
+
+amethyst_assets::register_format!("IMAGE", ImageFormat as TextureData);
+impl Format<TextureData> for ImageFormat {
+    fn name(&self) -> &'static str {
+        "IMAGE"
+    }
+
+    fn import_simple(&self, bytes: Vec<u8>) -> Result<TextureData, Error> {
+        load_from_image(std::io::Cursor::new(&bytes), self.0.clone())
+            .map(|builder| builder.into())
+            .map_err(|e| e.compat().into())
+    }
+}
+
+/// `PrefabData` for loading `Texture`s.
+///
+/// Will not add any `Component`s to the `Entity`, will only return a `Handle`
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(bound = "")]
+pub enum TexturePrefab {
+    /// Texture data
+    Data(TextureData),
+
+    // Generate texture
+    Generate(TextureGenerator),
+    /// Load file with format
+    File(String, Box<dyn Format<TextureData>>),
+
+    /// Clone handle only
+    #[serde(skip)]
+    Handle(Handle<Texture>),
+    /// Placeholder during loading
+    #[serde(skip)]
+    Placeholder,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub enum TextureGenerator {
+    Srgba(f32, f32, f32, f32),
+    LinearRgba(f32, f32, f32, f32),
+    //LinearRgbaFloat(f32, f32, f32, f32),
+    SrgbaCorners([(f32, f32, f32, f32); 4], Filter),
+}
+
+fn simple_builder<A: AsPixel>(data: Vec<A>, size: Size, filter: Filter) -> TextureData {
+    TextureBuilder::new()
+        .with_kind(Kind::D2(size, size, 1, 1))
+        .with_view_kind(ViewKind::D2)
+        .with_data_width(size)
+        .with_data_height(size)
+        .with_sampler_info(hal::image::SamplerInfo::new(
+            filter,
+            hal::image::WrapMode::Clamp,
+        ))
+        .with_data(data)
+        .into()
+}
+
+impl TextureGenerator {
+    fn data(&self) -> TextureData {
```

Comment:
could this be made `pub`, so that other code can also use the generator without a prefab?